packages/server/lib/routes.ts

```diff
@@ -219,8 +220,8 @@ publicAPI.route('/connection/metadata').post(apiAuth, postPublicMetadata);
 publicAPI.route('/connection/metadata').patch(apiAuth, patchPublicMetadata);
 publicAPI.route('/connection').post(apiAuth, connectionController.createConnection.bind(connectionController));
 publicAPI.route('/environment-variables').get(apiAuth, environmentController.getEnvironmentVariables.bind(connectionController));
-publicAPI.route('/sync/deploy').post(apiAuth, postDeploy);
-publicAPI.route('/sync/deploy/confirmation').post(apiAuth, postDeployConfirmation);
+publicAPI.route('/sync/deploy').post(apiAuth, cliMinVersion('0.39.25'), postDeploy);
+publicAPI.route('/sync/deploy/confirmation').post(apiAuth, cliMinVersion('0.39.25'), postDeployConfirmation);
```

Comment:
that's a breaking change. Would it make sense to first figure out if people are using older version to see how much of a problem is it going to be?

Reply:
The selected version is from 6months ago, I can remove if we are not confident 👌🏻

Reply:
the safest route would be to only add the middleware in `logging-mode` and let it run for a couple of days to see if any customer would be blocked by it.

Reply:
I'll log you are right 👌🏻